src/fromager/metrics.py

```diff
@@ -0,0 +1,77 @@
+import functools
+import logging
+import time
+import typing
+from collections import defaultdict
+from datetime import timedelta
+
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+_time_store: dict[str, dict[str, float]] = defaultdict(dict[str, float])
+_time_description_store: dict[str, str] = {}
```

Comment:
I would prefer to store this on the context. All functions should have a `ctx` argument.